From f77da2043c0840412f1a5bc89d04a8f595cc2027 Mon Sep 17 00:00:00 2001
From: Richard Leitner <richard.leitner@skidata.com>
Date: Fri, 1 Jun 2018 14:26:15 +0200
Subject: [PATCH 5/7] several autoheader define fixes

This patch was taken from Debian's libmxml 2.11-1 source:
	06_autoheader-fixes.patch
---
 configure.ac | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 6b3bbfa..47063dc 100644
--- a/configure.ac
+++ b/configure.ac
@@ -28,7 +28,7 @@ AC_CONFIG_HEADER(config.h)
 dnl Version number...
 VERSION="AC_PACKAGE_VERSION"
 AC_SUBST(VERSION)
-AC_DEFINE_UNQUOTED(MXML_VERSION, "Mini-XML v$VERSION")
+AC_DEFINE_UNQUOTED(MXML_VERSION, "Mini-XML v$VERSION", "MXML VERSION")
 
 dnl Clear default debugging options and set normal optimization by
 dnl default unless the user asks for debugging specifically.
@@ -113,7 +113,7 @@ AC_CACHE_CHECK(for long long int, ac_cv_c_long_long,
 	fi])
 
 if test $ac_cv_c_long_long = yes; then
-	AC_DEFINE(HAVE_LONG_LONG)
+	AC_DEFINE(HAVE_LONG_LONG, 1, "long long int support")
 fi
 
 dnl EPUB support (via libz and zipc)
@@ -122,7 +122,7 @@ AC_SUBST(MXML_EPUB)
 ZIPC=""
 AC_SUBST(ZIPC)
 AC_SEARCH_LIBS(gzgets,z,[
-	AC_DEFINE(HAVE_ZLIB_H)
+	AC_DEFINE(HAVE_ZLIB_H, 1, "<zlib.h> present")
 	ZIPC="zipc.o"
 	MXML_EPUB="mxml.epub"
 	LIBS="-lz $LIBS"])
@@ -135,7 +135,7 @@ PTHREAD_FLAGS=""
 PTHREAD_LIBS=""
 
 if test "x$enable_threads" != xno; then
-	AC_CHECK_HEADER(pthread.h, AC_DEFINE(HAVE_PTHREAD_H))
+	AC_CHECK_HEADER(pthread.h, AC_DEFINE(HAVE_PTHREAD_H, 1, "pthreads headers available"))
 
 	if test x$ac_cv_header_pthread_h = xyes; then
 		dnl Check various threading options for the platforms we support
-- 
2.11.0

